test/integration-tests/IntegrationTests.Helpers/TestHelper.cs

```diff
@@ -56,6 +57,36 @@ public Process StartSample(int traceAgentPort, string arguments, string packageV
                 processToProfile: executable);
         }
 
+        public ProcessResult RunSampleAndWaitForExit(int traceAgentPort, int? statsdPort = null, string arguments = null, string packageVersion = "", string framework = "", int aspNetCorePort = 5000)
+        {
+            var process = StartSample(traceAgentPort, arguments, packageVersion, aspNetCorePort: aspNetCorePort, statsdPort: statsdPort, framework: framework);
+
+            using var helper = new ProcessHelper(process);
+
+            process.WaitForExit();
```

Comment:
what happens if the process hangs? how about having some default timeout?

Reply:
I set process response timeout to 30 minutes (same for standard and error output). Whole process default timeout is now 60 minutes. 

Although I find the response checking timeout bit weird... Shouldn't it actually set both mutexes? In case of successful standard output, error mutex timeout is still set and could fire false positive.

Reply:
> error mutex timeout is still set and could fire false positive.

that would be bad 😋

Reply:
> I set process response timeout to 30 minutes (same for standard and error output). Whole process default timeout is now 60 minutes.

Then you should do: `timeout /= 2` otherwise it is very confusing

Reply:
refactored, to make sure that wait timeout is equally distributed.